pkg/controller/kogitoapp/resource/build_config_s2i.go

```diff
@@ -55,7 +55,7 @@ var (
 
 // newBuildConfigS2I creates a new build configuration for source to image (s2i) builds
 func newBuildConfigS2I(kogitoApp *v1alpha1.KogitoApp) (buildConfig buildv1.BuildConfig, err error) {
-	if kogitoApp.Spec.Build == nil || kogitoApp.Spec.Build.GitSource == nil {
+	if kogitoApp.Spec.Build == nil || &kogitoApp.Spec.Build.GitSource == nil {
```

Comment:
Since the GitSource is not a pointer any more, it cannot be nil, right?
Therefore the `&kogitoApp.Spec.Build.GitSource == nil` is not needed.

Reply:
Wrong, I  can't guarantee that `GitSource` won't be blank, I wrote a test to validate this, will update the PR soon. Thanks for the heads up!